src/app/modules/ix-table2/components/ix-table-columns-selector/ix-table-columns-selector.component.ts

```diff
@@ -63,8 +68,13 @@ export class IxTableColumnsSelectorComponent<T = unknown> implements OnChanges {
 
   private setInitialState(): void {
     this.columns = _.cloneDeep(this.defaultColumns);
-    this.hiddenColumns.clear();
-    this.hiddenColumns.select(...this.columns.filter((column) => column.hidden));
+    this.hiddenColumns.select(...this.columns);
+
+    this.defaultColumns
+      .filter((column) => !column.hidden)
+      .map((column) => this.defaultColumns.findIndex((column2) => column.title === column2.title))
```

Comment:
Can we rewrite this in some other way, this is a bit of hard to follow.

Reply:
> Can we rewrite this in some other way, this is a bit of hard to follow.

sure, but only `findIndex` works, that's weird